ts/component/tab-bar-builder.ts

```diff
@@ -0,0 +1,129 @@
+import { Pair } from "../util/pair";
+import { getSearchParams, updateUrlParams } from "../util/urls";
+import { TableBuilder } from "./table-builder";
+
+class Tab {
+  tabContainer: HTMLButtonElement;
+  table: Pair<any, string>;
+  selected: boolean = false;
+  onSelect: (table: Pair<any, string>) => void;
```

Comment:
Remove `table` field and argument from `onSelect` function. The only time `table` is used is when calling `onSelect`, but these are both provided in the constructor, so the caller already has access to the table to use within `onSelect`. The tab bar could also be reused for other purposes not involving tables